cmd/zrepl/zrepl.c

```diff
@@ -218,6 +219,12 @@ uzfs_zvol_io_receiver(void *arg)
 			continue;
 		}
 
+		if (zio_cmd->hdr.opcode == ZVOL_OPCODE_WRITE) {
+			(void) pthread_mutex_lock(&zinfo->zinfo_mutex);
+			zinfo->running_io_seq = zio_cmd->hdr.io_seq;
+			(void) pthread_mutex_unlock(&zinfo->zinfo_mutex);
```

Comment:
maybe we could use a more lightweight mechanism to atomically update the 64bit number without locking zinfo. Perhaps there are functions which could help us. Like compare-and-swap or something similar.

Reply:
I tried but could not found. Will take this as part of perfromance optimization.